src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -1137,10 +1137,41 @@ abstract class RefChecks extends Transform {
         else warnIfLubless()
       }
     }
+
+    private def checkSensibleAnyEquals(pos: Position, qual: Tree, name: Name, sym: Symbol, other: Tree) = {
+      def underlyingClass(tp: Type): Symbol = {
+        val sym = tp.widen.typeSymbol
+        if (sym.isAbstractType) underlyingClass(sym.info.upperBound)
+        else sym
+      }
+      val receiver = underlyingClass(qual.tpe)
+      val actual   = underlyingClass(other.tpe)
+      def typesString = normalizeAll(qual.tpe.widen)+" and "+normalizeAll(other.tpe.widen)
+      def nonSensiblyEquals() = {
+        reporter.warning(pos, s"comparing values of types $typesString using `${name.decode}` is unsafe due to cooperative equality; use `==` instead")
```

Comment:
This should call `refchecksWarning`, going directly do `reporter.warning` bypasses the `Wconf` infrastructure. Any takers?

```
➜  sandbox git:(2.13.x) qs '-Wconf:msg=statement:e,msg=equality:e'
Welcome to Scala 2.13.4-20201009-113655-8fbebaa (Java HotSpot(TM) 64-Bit GraalVM EE 19.2.1, Java 1.8.0_231).
Type in expressions for evaluation. Or try :help.

scala> def f = { 1; 2 }
                 ^
       error: a pure expression does nothing in statement position; multiline expressions might require enclosing parentheses

scala> def g = 1L equals 1
                  ^
       warning: comparing values of types Long and Int using `equals` is unsafe due to cooperative equality; use `==` instead
def g: Boolean
```

Reply:
I just noticed this myself, trying to fix up the warning for #9235

Reply:
@lrytz if all it is is changing that one line, I'll take a stab

Reply:
@lrytz what's that `qs` command/alias?

Reply:
Likely `build/quick/bin/scala` or perhaps `$(git root 2>/dev/null)/build/quick/bin/scala`

Reply:
> `$(git root 2>/dev/null)/build/quick/bin/scala`

If only I was that smart. Though `git root` [doesn't seem to be a built-in](https://stackoverflow.com/questions/957928/is-there-a-way-to-get-the-git-root-directory-in-one-command). Adopted, thank you.